api/graphite.go

```diff
@@ -101,63 +209,62 @@ func (s *Server) renderMetrics(ctx *middleware.Context, request models.GraphiteR
 	}
 
 	reqs := make([]models.Req, 0)
-	parsedTargets := make(map[string]string)
+
+	// consolidatorForPattern[<pattern>]<consolidateBy>
+	consolidatorForPattern := make(map[string]string)
 	patterns := make([]string, 0)
 	type locatedDef struct {
 		def  schema.MetricDefinition
 		node *cluster.Node
 	}
 
+	//locatedDefs[<pattern>][<def.id>]locatedDef
 	locatedDefs := make(map[string]map[string]locatedDef)
-	queryForTarget := make(map[string]string)
-	for _, query := range targets {
-		target, consolidateBy, err := parseTarget(query)
+	//targetForPattern[<pattern>]<target>
+	targetForPattern := make(map[string]string)
+	for _, target := range targets {
+		pattern, consolidateBy, err := parseTarget(target)
 		if err != nil {
 			ctx.Error(http.StatusBadRequest, err.Error())
 			return
 		}
-		parsedTargets[target] = consolidateBy
-		patterns = append(patterns, target)
-		queryForTarget[target] = query
-		locatedDefs[target] = make(map[string]locatedDef)
-
-		// metricDefs only get updated periodically, so we add a 1day (86400seconds) buffer when
-		// filtering by our From timestamp.  This should be moved to a configuration option,
-		// but that will require significant refactoring to expose the updateInterval used
-		// in the MetricIdx.
-		seenAfter := int64(fromUnix)
-		if seenAfter != 0 {
-			seenAfter -= 86400
-		}
-		nodes, err := s.MetricIndex.Find(ctx.OrgId, target, seenAfter)
-		if err != nil {
-			response.Write(ctx, response.NewError(http.StatusBadRequest, err))
-			return
-		}
-		for _, node := range nodes {
-			for _, def := range node.Defs {
-				locatedDefs[target][def.Id] = locatedDef{def, cluster.ThisNode}
+		consolidatorForPattern[pattern] = consolidateBy
+		patterns = append(patterns, pattern)
+		targetForPattern[pattern] = target
+		locatedDefs[pattern] = make(map[string]locatedDef)
+
+	}
+
+	rSeries, err := s.findSeries(ctx.OrgId, patterns, int64(fromUnix))
+	if err != nil {
+		response.Write(ctx, response.NewError(http.StatusInternalServerError, err))
```

Comment:
note that err here can contain low-level errors revealing details about our infrastructure. (could be like `dial timeout to 10.240.x.x ` , `network unreachable`, etc)
 I think those should be abstracted away to the user, and just tell the user we had an internal problem with the cluster, or something